tests/lib/KeyStore.spec.js

```diff
@@ -157,12 +155,51 @@ describe('KeyStore', () => {
 
         expect(migrationCookieDeleted && accountsCookieDeleted).toBe(true);
         const key1 = await KeyStore.instance._get(Dummy.keyInfos[0].id);
-        expect(key1).toEqual(Dummy.keyRecords[0]);
+        expect(key1).toEqual(Dummy.storedKeyRecords()[0]);
 
         // TODO: Expect Accounts DB to not be deleted
 
         await Dummy.Utils.deleteDummyAccountStore();
     });
 
+    it('doesn\'t store same key twice', async () => {
+        // first clear database
+        await Dummy.Utils.deleteDummyKeyStore();
+
+        let currentKeys = await KeyStore.instance.list();
+
+        // add key
+        const passphrase = Nimiq.BufferUtils.fromAscii(Dummy.encryptionPassword);
+        await KeyStore.instance.put(new Key(Dummy.secrets[1]), passphrase);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(1);
+
+        // add key again
+        await KeyStore.instance.put(new Key(Dummy.secrets[1]), passphrase);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(1);
+
+        // add key again with different password
+        const passphrase2 = Nimiq.BufferUtils.fromAscii(Dummy.encryptionPassword2);
```

Comment:
I suggest moving the "generation" of the password buffers to the top together and naming them `password1/2` instead of passphrase.